cpp/src/arrow/type.h

```diff
@@ -1968,6 +1968,8 @@ class ARROW_EXPORT Schema : public detail::Fingerprintable,
   Result<std::shared_ptr<Schema>> SetField(int i,
                                            const std::shared_ptr<Field>& field) const;
 
+  Result<std::shared_ptr<Schema>> WithNames(const std::vector<std::string>& names) const;
```

Comment:
What does this function do? (Missing docstring)

Reply:
It gives a schema with the names changed, similar to other `With*` methods. I could add a docstring.